### PR TITLE
Epic: Fields API and Legacy Form Consumer

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@since` tags included in DocBlocks
-   [ ] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@kjohnson I’ve noticed, when building fields in PhpStorm that the autocomplete is offering a lot of the trait properties. Please update these to be protected. Until such a time as we can use typed properties in PHP 7.4 I don’t trust properties to be directly mutated. Also, I like the declarative abstraction of the methods to how the settings are applied and it keeps things fluent.

**Comment by kjohnson** [comment]
@JasonTheAdams yeah, that makes sense. I usually start with `public` and then update as needed. This makes sense for these to be protected to enforce the fluent methods.

Resolved by https://github.com/impress-org/givewp/pull/5568/commits/34786327bbd3d0e32b3e3e62b674a6507d806d01.

**Comment by JasonTheAdams** [comment]
> @JasonTheAdams yeah, that makes sense. I usually start with `public` and then update as needed. This makes sense for these to be protected to enforce the fluent methods.

Ah, that's interesting. I was taught to do the opposite: keep scope as narrow as possible, opening only as necessary.

File: src/Form/LegacyConsumer/templates/checkbox.html.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<input
+	type="checkbox"
+	class="give-input required"
+	name="give_<?php echo $field->getName(); ?>"
+	id="give-<?php echo $field->getName(); ?>"
+	<?php if ( $field->isRequired() ) : ?>    
+		required="" aria-required="true"
+	<?php endif; ?>
+	tabindex="1"
```
Comment (ravinderk) [inline src/Form/LegacyConsumer/templates/checkbox.html.php:9]: @kjohnson is there any reason to hardcode tab index to `1` in all `*.html.php`

Comment (kjohnson) [reply src/Form/LegacyConsumer/templates/checkbox.html.php:9]: @ravinderk yes, it is added for accessibility.

**Comment by kjohnson** [comment]
> > > I’ve noticed, when building fields in PhpStorm, that the autocomplete is offering a lot of the trait properties. Please update these to be protected. Until such a time as we can use typed properties in PHP 7.4 I don’t trust properties to be directly mutated. Also, I like the declarative abstraction of the methods to how the settings are applied and it keeps things fluent.
> > yeah, that makes sense. I usually start with `public` and then update as needed. This makes sense for these to be protected to enforce the fluent methods.
> 
> Ah, that's interesting. I was taught to do the opposite: keep scope as narrow as possible, opening only as necessary.

I think it comes down to the scope of extensibility.

In my experience with WordPress there are a lot of requests to open things up, which introduces community management overhead. Starting with `public` reduces friction for a community that is often changing this to fit their use case. I have a few anecdotes extending other WordPress products that required large workarounds for seemingly simple needs, in which the developer unnecessarily locked things down. This creates a problem when you are extending something of which you don't control the source and extending it in a sustainable way.

This can be in great contrast to a closed source product in which the only developers working on the project have the access to open up what they need, as they need it. In this case, you start with `protected` and make the case for `public` only when necessary.

Also, Jeffery Way makes a point about allowing developers to use "sharp tools" and not needing to "protect developers from themselves". It is an interesting philosophy and one which I think has its applications.

Your point of IDE autocomplete is interesting. Building out a first-class API is certainly the  "better" approach and this limiting public properties makes sense.

**Comment by kjohnson** [comment]
@ravinderk thanks for catching that typo! Not sure how that slipped through...

**Pull request merged**